### DISCUSSION TITLE
Feature request: pull_request base branch change event

### DISCUSSION THREAD

**Discussion by stefanfoulis** [community/community#35058] [Actions]
Currently it is possible to react to the event when a different base branch is assigned to a PR with the `edited` event like so:
```
on:
  pull_request:
    types: [edited]
```
However this event also fires if the PR `title` or `description` changes. And it does _not_ fire if new commits are added to the `base` branch.

My suggestion is to add a new event, called something like `base_changed`, that will fire if a different base branch is assigned to the PR or if the `head` of the base branch changes.

```
on:
  pull_request:
    types: [base_changed]
```

In my case I'd like to use it to re-run CI on a PR using the merge commit if the base changes. I know there are github actions that attempt this, but they all have to resort to workarounds to achieve this. E.g by adding an empty commit to the PR. And they need a _Personal Access Token_ to work.
Another workaround could be to use `workflow_dispatch` or `workflow_run` and put all the CI logic in those. But when dispatching that shared workflow from various events we loose the "PR context" and all the associated automatic commit statuses. Overall these other approaches are all very cumbersome.

**Comment by willbush** [comment]
This would be useful for [an action that runs on every PR](https://github.com/NixOS/nixpkgs/blob/d36cb226a5ad228d1e6a7f2e3d635964d4193772/.github/workflows/check-by-name.yml#L17) in [nixpkgs](https://github.com/NixOS/nixpkgs). more context: https://github.com/NixOS/nixpkgs/pull/313345

**Comment by JohnRTitor** [comment]
Would love to get this implemented.

**Comment by infinisil** [comment]
Duplicate of the newer https://github.com/orgs/community/discussions/64119
